lib/manager/gradle-lite/extract.ts

```diff
@@ -13,6 +13,21 @@ import {
   toAbsolutePath,
 } from './utils';
 
+// Enables reverse sorting in generateBranchConfig()
+//
+// Required for grouped dependencies to be upgraded
+// correctly in single branch.
+//
+// https://github.com/renovatebot/renovate/issues/8224
+function elevateFileReplacePositionField(
+  deps: PackageDependency<ManagerData>[]
+): PackageDependency<ManagerData>[] {
+  return deps.map((dep) => ({
+    ...dep,
+    fileReplacePosition: dep?.managerData?.fileReplacePosition,
```

Comment:
Why not make sure we use `managerData` everywhere? Because I'm not sure this field will be available from all the places where `fileReplacePosition` is needed. Potentially, maybe it can be pretty big refactoring for the thing which is pretty hacky.